infrastructure/persistence/file.go

```
@@ -159,34 +163,113 @@ func (l *ltsvReader) labelAndData(s string) (string, string, error) {
+	return gzReader, newCloser, nil
+}
+
+func wrapBZ2Reader(reader io.Reader, closer func() error) (io.Reader, func() error, error) {
+	bz2Reader := bzip2.NewReader(reader)
+	return bz2Reader, closer, nil
+}
+
+func wrapXZReader(reader io.Reader, closer func() error) (io.Reader, func() error, error) {
+	xzReader, err := xz.NewReader(reader)
+	if err != nil {
+		closer()
```

golangci-lint (1.64.8)

Error return value is not checked

(errcheck)